msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/mam/TestCase2798415.kt

```diff
@@ -0,0 +1,148 @@
+import com.microsoft.identity.labapi.utilities.constants.UserType
+import com.microsoft.identity.labapi.utilities.exception.LabApiException
+import org.junit.Test
+
+// Shared device mode - TrueMAM: Sign In with Teams and then SignOut and Sign Back In 
+// https://identitydivision.visualstudio.com/Engineering/_workitems/edit/2798415
+@SupportedBrokers(brokers = [BrokerMicrosoftAuthenticator::class])
+@RetryOnFailure
+class TestCase2798415 : AbstractMsalBrokerTest() {
+    private val TAG = TestCase2798415::class.java.simpleName
+    @Test
+    @Throws(MsalException::class, InterruptedException::class, LabApiException::class)
+    fun test_2798415() {
+        val adminUserLabQuery = getAdminAccountLabQuery()
+
+        // get username and password for this account
+        val admin: ILabAccount = mLabClient.getLabAccount(adminUserLabQuery)
+        Logger.i(TAG, "Performing Shared Device Registration.")
+        mBroker.performSharedDeviceRegistration(admin.username, admin.password)
+
+        // install CP
+        val companyPortal = BrokerCompanyPortal()
+        companyPortal.install();
+
+        // install Teams.
+        val teams = TeamsApp(LocalApkInstaller())
+        teams.install()
+        teams.launch()
+        teams.handleFirstRun()
+
+        val username = mLabAccount.username
+        val password = mLabAccount.password
+
+        val teamsPromptHandlerParameters = FirstPartyAppPromptHandlerParameters.builder()
+            .prompt(PromptParameter.SELECT_ACCOUNT)
+            .loginHint(username)
+            .broker(mBroker)
+            .registerPageExpected(false)
+            .enrollPageExpected(false)
+            .consentPageExpected(false)
+            .speedBumpExpected(false)
+            .build()
+
+        // Sign in the first time
+        teams.addFirstAccount(username, password, teamsPromptHandlerParameters)
+        // handle app protection policy in CP i.e. setup PIN when asked
+        (companyPortal as IMdmAgent).handleAppProtectionPolicy()
+        teams.onAccountAdded()
+        teams.forceStop() // Teams sometimes seems to like to pop up on screen randomly
+
+        teams.signOut()
+
+        // Sign in again
+        val teamsPromptHandlerParameters2 = FirstPartyAppPromptHandlerParameters.builder()
+            .prompt(PromptParameter.SELECT_ACCOUNT)
+            .loginHint(username)
+            .broker(mBroker)
+            .registerPageExpected(false)
+            .enrollPageExpected(false)
+            .consentPageExpected(false)
+            .speedBumpExpected(false)
+            .secondPasswordPageExpected(true)
+            .build()
+
+        try {
+            teams.addFirstAccount(username, password, teamsPromptHandlerParameters2)
+        } catch (e : AssertionError) {
+            if (e.message == "Prompt handler failed to handle second password prompt...") {
+                // We can continue if we fail on the second password prompt, as it seems to not always come up
```

Comment:
Is this a bug on Teams' side? why would we expect users to enter pwd twice?

Reply:
Seems to be expected from test steps
![image](https://github.com/AzureAD/microsoft-authentication-library-for-android/assets/88730756/e88a42c5-3ae2-4e4c-b8f7-3801e760ba65)

Reply:
The steps are copied over form an existing test case. Hence this change.